### ISSUE TITLE
HHVM reported Siege bugs in >2.78 ?

### ISSUE THREAD

**Issue by centminmod** [JoeDog/siege#15]
Hi curious if the bugs reported at http://hhvm.com/blog/9293/lockdown-results-and-hhvm-performance for Siege >2.78 been fixed in 3.1.0 yet ?

> Siege — When benchmarking frameworks we used Siege 2.78, currently all versions of Siege 3 have known problems. In particular Siege 3.0.0 to 3.0.7 send incorrect HOST headers to ports other than 80, and 443, and Siege 3.0.8 and 3.0.9 occasionally send bad paths due to incorrect redirect handling.

**Comment by JoeDog** [comment]
I was not aware of that post so I'm not sure what they're talking about.
The port issue is fixed. I'm not sure what they're referring to when they
say it sends incorrect paths. I've entertained discussions about this with
others. I claim siege complies with the RFC.

Jeff

Hi curious if the bugs reported at
http://hhvm.com/blog/9293/lockdown-results-and-hhvm-performance for Siege
>2.78 been fixed in 3.1.0 yet ?

Siege — When benchmarking frameworks we used Siege 2.78, currently all
versions of Siege 3 have known problems. In particular Siege 3.0.0 to 3.0.7
send incorrect HOST headers to ports other than 80, and 443, and Siege
3.0.8 and 3.0.9 occasionally send bad paths due to incorrect redirect
handling.

—
Reply to this email directly or view it on GitHub
<https://github.com/JoeDog/siege/issues/15>.

**Comment by centminmod** [comment]
@JoeDog yeah i do not know what they mean about incorrect paths either so thought I'd ask :)

Just want to say love your work on Siege - been a long time fan and user http://wordpress7.centminmod.com/122/wordpress-super-cache-benchmark-siege-http-load-testing/ and love that you now have the code on Github :)

cheers

George

**Comment by JoeDog** [comment]
Thanks for the kind words, George. I appreciate it.

J.

**Comment by JoeDog** [comment]
George,

I also like your entry on testing wp-supercache. Good stuff.

J.

**Comment by centminmod** [comment]
thanks Jeff :)

**Comment by centminmod** [comment]
@JoeDog Jeff some more info on their encounted issues in Siege 3.x https://github.com/hhvm/oss-performance/issues/48

**Comment by fredemmott** [comment]
I can't reproduce issues in 3.1.0; I'm keeping 3.x blacklisted in oss-performance for now until we have time to check that 3.1.0 gives similar results to 2.78 across the suite - average byte size of response, response code ratios, etc - this is a fairly time-consuming process, which we also go through whenever we benchmark a new version of PHP, HHVM, or some framework - not specific to Siege.

**Comment by fredemmott** [comment]
Filed hhvm/oss-performance#50 to investigate changing the blacklist from 3.* to 3.0.*
